File: Code/Haskell/problem003/euler003.hs

theNumber :: Integer
theNumber = 600851475143

startingPoint n = floor $ sqrt $ fromIntegral n 

beginAt = startingPoint theNumber

isPrime :: Integer -> Bool
isPrime n = calcIsPrime n (n - 1)

calcIsPrime :: Integer -> Integer -> Bool
calcIsPrime n 1 = True
calcIsPrime n m = case n `rem` m == 0 of
                    True -> False
                    False -> calcIsPrime n (m-1)

euler n test 
 | test <= 1 = 0
 | n `rem` test == 0 && isPrime n = n
 | otherwise = euler n (test - 1)

main = print $ euler theNumber beginAt

File: Code/Python/problem003/euler003.py
#!/usr/bin/env python
from __future__ import print_function
import math

theNumber = 600851475143


def startingPoint(num):
    start = math.floor(math.sqrt(num))
    print("The starting point is {0}".format(start))
    return start


def isprime(num):
    factor = num - 1
    while (factor > 1):
        if num % factor == 0:
            return False
        factor = factor - 1
    return True


def euler(num, test):
    while test > 0:
        if num % test == 0 and isprime(test):
            print("The largest prime factor is {0}".format(test))
            return test
        test = test - 1


if __name__ == '__main__':
    euler(theNumber, startingPoint(theNumber))
